ui/ducks/send/swap-and-send-utils.ts

```diff
@@ -0,0 +1,198 @@
+import { isNumber } from 'lodash';
+import {
+  SWAPS_API_V2_BASE_URL,
+  SWAPS_CLIENT_ID,
+} from '../../../shared/constants/swaps';
+import { SECOND } from '../../../shared/constants/time';
+import fetchWithCache from '../../../shared/lib/fetch-with-cache';
+import {
+  addHexPrefixToObjectValues,
+  truthyDigitString,
+  truthyString,
+  validHex,
+  validateData,
+} from '../../../shared/lib/swaps-utils';
+import { decimalToHex } from '../../../shared/modules/conversion.utils';
+import { isValidHexAddress } from '../../../shared/modules/hexstring-utils';
+
+type ADDRESS = `0x${string}`;
+
+type Request = {
+  chainId: number; // chain ID as decimal
+  sourceAmount: string; // big number string
+  sourceToken: ADDRESS;
+  destinationToken: ADDRESS;
+  sender: ADDRESS;
+  recipient: ADDRESS;
+  slippage: string; // slippage as a percentage; e.g. '1.0' for 1%
+};
+
+type Quote = {
+  gasParams: {
+    maxGas: number;
+  };
+  trade: {
+    data: string;
+    to: string;
+    from: string;
+    value: string;
+  };
+  approvalNeeded: null | {
+    data: string;
+    to: string;
+    from: string;
+  };
+  sourceAmount: string;
+  destinationAmount: string;
+  sourceToken: string;
+  destinationToken: string;
+  sender: string;
+  recipient: string;
+  aggregator: string;
+  aggregatorType: string;
+  error: null | object;
+  fee: number;
+};
```

Comment:
Does swaps or their SDK export these? if not, and these are the same types used by swaps, maybe these would be better in a *.d.ts file?

Reply:
We can't count on the typing being the same since we use a different endpoint (see: [comment](https://github.com/MetaMask/metamask-extension/pull/23835#discussion_r1566173908))